### PR TITLE
(feat) Various UI mods

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR applies various UI refactors to the Medication Dispensing UI, including:

- Fixing the thick border between the table rows (by setting `display: none` on the collapsed `TableExpandedRow`).
- Fixing the double top border on the pagination container of the datatable.
- Adding an empty state to the DataTable when the search term doesn't yield any matching prescriptions.
-  Fixing the translation string in the `Prescription details` Allergies tile to account for plurals.

Other changes include:

- Enabling linting for stylesheets (an oversight from the past). This explains the large diff owing from running Prettier on all the stylesheets.
- Enabling linting for tests

### Before

CleanShot 2024-08-22 at 11  53 43@2x

CleanShot 2024-08-23 at 12  00 45@2x

CleanShot 2024-08-23 at 12  02 48@2x

### After

CleanShot 2024-08-22 at 11  53 56@2x

CleanShot 2024-08-23 at 12  01 12@2x

CleanShot 2024-08-23 at 12  03 07@2x

### CODE REVIEW

File: src/forms/medication-dispense-review.component.tsx [hunk 1]
```diff
@@ -193,301 +193,303 @@ const MedicationDispenseReview: React.FC<MedicationDispenseReviewProps> = ({
...
-                reference: 'Medication/' + selectedItem?.id,
-                display: getOpenMRSMedicineDrugName(selectedItem),
-              },
-            });
-            setIsEditingFormulation(false);
-          }}
-          required
-        />
-      )}
+      <Stack gap={5}>
```
Comment (denniskigen) [inline src/forms/medication-dispense-review.component.tsx:196]: This is for consistent spacing between the form elements.

File: src/prescriptions/prescription-tab-panel.component.tsx [hunk 2]
```diff
@@ -100,33 +101,51 @@ const PrescriptionTabPanel: React.FC<PrescriptionTabPanelProps> = ({ searchTerm,
...
+                  <Tile className={styles.filterEmptyStateTile}>
+                    <p className={styles.filterEmptyStateContent}>
+                      {t('noPrescriptionsToDisplay', 'No prescriptions to display')}
+                    </p>
+                    <p className={styles.filterEmptyStateHelper}>{t('checkFilters', 'Check the filters above')}</p>
+                  </Tile>
+                </Layer>
+              </div>
+            )}
+            {prescriptionsTableRows?.length > 0 && (
```
Comment (denniskigen) [inline src/prescriptions/prescription-tab-panel.component.tsx:134]: So the pagination container is hidden in the empty state.

**Review by mogoodrich — APPROVED** [review]
Thanks @denniskigen !

It's hard for me to review with all the formatting changes, but I trust you... :)

Why all the formatting changes, is yarn lint not set up properly?

**Comment by ibacher** [comment]
> Why all the formatting changes, is yarn lint not set up properly?

I think it mostly derives from the fact that we used to have prettier ignore most things that weren't Typescript files, and we've been changing things so that we use it to reformat Sass files as well.

**Review by ibacher — APPROVED** [review]

**Pull request merged**